reporter/src/main/kotlin/reporters/SpdxDocumentReporter.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.SpdxDocumentModelMapper
+import org.ossreviewtoolkit.spdx.SpdxModelSerializer
+
+import java.io.File
+
+/**
+ * Creates YAML and JSON SPDX documents.
+ */
+class SpdxDocumentReporter : Reporter {
+    companion object {
+        const val CREATION_INFO_COMMENT = "document.comment"
+        const val DOCUMENT_COMMENT = "document.comment"
+        const val DOCUMENT_NAME = "document.name"
+        const val OUTPUT_FILE_FORMATS = "output.file.formats"
+
+        private const val DOCUMENT_NAME_DEFAULT_VALUE = "Unnamed document"
+    }
+
+    enum class FileFormat(val reportFilename: String) {
+        JSON("spdx-document.spdx.json"),
+        YAML("spdx-document.spdx.yml");
```

Comment:
Shouldn't we better use use `package(s).spdx.*` as discussed at https://github.com/spdx/spdx-spec/issues/439, or maybe even use the root / first project name?

Reply:
I like `spdx-document` but am ok with `package` or `packages. @tsteenbe @mnonnenmacher any preference from your side?

Reply:
From the SPDX spec it's a SPDX document that holds the info but a SPDX document does not have to contain a package - it can only contain info on files or snippets so I would not use `package` as the default name. We use `package` in https://github.com/spdx/spdx-spec/issues/439 as we for this use case we specifically want to describe a package and using the`package` is common in other package manager (think `package.json` for npm).

I would only use the root / first project name by default if we can find out a way to make it a save filename that is accepted by various operating system.

I would propose to A) use the defaults as @fviernau  but provide an option for user to specify name of output file or B) use the root / first project name by default but ensure it results in a 'safe' [a-z|A-Z|0-9|-] filename.

Reply:
I also don't like the "spdx" duplication in "spdx-document.spdx.*" so much. Maybe just "document.spdx.*" is anther option. Let's see what the others say.

Reply:
@tsteenbe our comments overlapped. So how do you like `document.spdx.*` here?

Reply:
I don't like using the project name as this logic makes the script calling the reporter more complex. Imo the name should be a no brainer to predict.

Reply:
> A) use the defaults as @fviernau but provide an option for user to specify name of output file

Actually, we don't support custom names for any of the other reporter, so I'm fine with hard-coding the name here.

Reply:
So are all fine with `document.spdx.*` (@tsteenbe) ?

Reply:
I just changed it to `document.spdx.*`.